proxy/http/HttpSM.cc

```diff
@@ -2357,6 +2356,9 @@ HttpSM::state_cache_open_write(int event, void *data)
 
   // Make sure we are on the "right" thread
   if (ua_txn) {
+    if (pending_action) {
+      pending_action->cancel();
```

Comment:
I'd be tempted to put a debug assert here, as really `pending_action` shouldn't be set.

Reply:
I did have an assert there initially.  It went off with some of the thread tests, so I changed it to the cancel.